internal/resources/metal/virtual_circuit/resource.go

```diff
@@ -354,19 +355,24 @@ func resourceMetalVirtualCircuitRead(ctx context.Context, d *schema.ResourceData
 	return diag.FromErr(errors.Join(errs...))
 }
 
+// Waiting for delete is different from waiting for a status, because we explicitly _want_
+// to get a 404 from the API on delete so that we know the resource is truly gone.
+// We may want to have a separate wait function for deletions so that it is easier
+// to understand the code
 func getVCStateWaiter(ctx context.Context, client *metalv1.APIClient, id string, timeout time.Duration, pending, target []string) *retry.StateChangeConf {
 	return &retry.StateChangeConf{
 		Pending: pending,
 		Target:  target,
 		Refresh: func() (interface{}, string, error) {
 			vc, resp, err := client.InterconnectionsApi.GetVirtualCircuit(ctx, id).Execute()
 			if err != nil {
-				if resp != nil {
-					// The resource delete function uses this waiter and relies
-					// on it to return an ErrorResponse error so it can treat
-					// a 404 as success.  This conversion is done here for now
-					// to avoid a larger refactoring.
-					err = equinix_errors.FriendlyErrorForMetalGo(err, resp)
+				// Per current usage, when waiting for delete we pass in an empty target
+				// list; therefore, if the target list is empty we assume that a 404 or 403
+				// from the API is a success rather than a failure.
+				if len(target) == 0 {
+					if equinix_errors.IgnoreHttpResponseErrors(http.StatusNotFound, http.StatusForbidden)(resp, err) == nil {
```

Comment:
I suspect that we've been ignoring far more HTTP 403 errors than we should.  My understanding is that there are limited times when we might want to ignore a 403 for a Metal device, but that other resources do not exhibit the same behavior: https://github.com/equinix/terraform-provider-equinix/issues/233

Reply:
I'm creating a separate issue for the above comment; the aim of this PR is to check for the same statuses, and there's already plenty of tech debt cleanup coming in through that narrow scope.